src/probability/independence.lean

```diff
@@ -630,6 +630,82 @@ begin
     rw [h_sets_s'_univ hi, set.univ_inter], },
 end
 
+lemma Indep_fun.indep_fun_prod [is_probability_measure μ]
+  {ι : Type*} {β : ι → Type*} {m : Π i, measurable_space (β i)}
+  {f : Π i, α → β i} (hf_Indep : Indep_fun m f μ) (hf_meas : ∀ i, measurable (f i))
+   (i j k : ι) (hik : i ≠ k) (hjk : j ≠ k) :
+  indep_fun (λ a, (f i a, f j a)) (f k) μ :=
+begin
+  classical,
+  have h_right : f k = (λ p : (Π j : ({k} : finset ι), β j), p ⟨k, finset.mem_singleton_self k⟩)
+    ∘ (λ a (j : ({k} : finset ι)), f j a) := rfl,
+  have h_meas_right : measurable
+      (λ p : (Π j : ({k} : finset ι), β j), p ⟨k, finset.mem_singleton_self k⟩),
+    from measurable_pi_apply ⟨k, finset.mem_singleton_self k⟩,
+  let s : finset ι := {i, j},
+  have h_left : (λ ω, (f i ω, f j ω))
+    = (λ p : (Π l : s, β l), (p ⟨i, finset.mem_insert_self i _⟩,
+        p ⟨j, finset.mem_insert_of_mem (finset.mem_singleton_self _)⟩))
+      ∘ (λ a (j : s), f j a),
+  { ext1 a,
+    simp only [prod.mk.inj_iff],
+    split; refl, },
+  have h_meas_left : measurable (λ p : (Π l : s, β l), (p ⟨i, finset.mem_insert_self i _⟩,
+      p ⟨j, finset.mem_insert_of_mem (finset.mem_singleton_self _)⟩)),
+    from measurable.prod (measurable_pi_apply ⟨i, finset.mem_insert_self i {j}⟩)
+      (measurable_pi_apply ⟨j, finset.mem_insert_of_mem (finset.mem_singleton_self j)⟩),
+  rw [h_left, h_right],
+  refine (hf_Indep.indep_fun_finset s {k} _ hf_meas).comp h_meas_left h_meas_right,
+  intros x hx,
+  simp only [finset.inf_eq_inter, finset.mem_inter, finset.mem_insert, finset.mem_singleton] at hx,
+  simp only [finset.bot_eq_empty, finset.not_mem_empty],
+  cases hx.1 with hx_eq hx_eq; rw hx_eq at hx,
+  { exact hik hx.2, },
+  { exact hjk hx.2, },
+end
+
+lemma Indep_fun.add [is_probability_measure μ]
```

Comment:
can you instead have `Indep_fun.mul` with an attribute `to_additive`, to have both versions at once?